src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -105,6 +108,21 @@
      */
     private final Cache<String, ActiveDirectoryGroupDetails> groupCache;
 
+    /**
+     * Sets com.sun.jndi.ldap.connect.timeout
+     */
+    private final Long ldapConnectTimeout;
+
+    /**
+     * Sets com.sun.jndi.ldap.read.timeout
+     */
+    private final Long ldapReadTimeout;
+
+    /**
+     * Properties of the current context
+     */
+    private Hashtable<String, String> props = new Hashtable<String, String>();
```

Comment:
I think you are over complicating this...  the provider should not care what the timeouts or properties are.  Keep them with the descriptor, as they are not used outside of the descriptor.